core/trino-main/src/main/java/io/trino/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -203,12 +212,19 @@ private Visitor(
                 dynamicFilter = dynamicFilterService.createDynamicFilter(session.getQueryId(), dynamicFilters, node.getAssignments(), typeProvider);
             }
 
+            Constraint constraint = filterPredicate
+                    .map(predicate -> filterConjuncts(metadata, predicate, expression -> !DynamicFilters.isDynamicFilter(expression)))
+                    .map(expression -> new LayoutConstraintEvaluator(metadata, typeAnalyzer, session, typeProvider, node.getAssignments(), expression))
+                    .map(evaluator -> new Constraint(TupleDomain.all(), evaluator::isCandidate, evaluator.getArguments()))
```

Comment:
add a comment
// we are interested only in functional predicate here so we set the summary to ALL.